src/main/java/io/confluent/kafkarest/SimpleConsumerManager.java

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.kafkarest;
+
+import io.confluent.kafka.serializers.KafkaAvroDecoder;
+import io.confluent.kafkarest.converters.AvroConverter;
+import io.confluent.kafkarest.entities.AvroConsumerRecord;
+import io.confluent.kafkarest.entities.BinaryConsumerRecord;
+import io.confluent.kafkarest.entities.ConsumerRecord;
+import io.confluent.kafkarest.entities.EmbeddedFormat;
+import io.confluent.kafkarest.simpleconsumerspool.NaiveSimpleConsumerPool;
+import io.confluent.kafkarest.simpleconsumerspool.SimpleConsumerPool;
+import io.confluent.kafkarest.simpleconsumerspool.SizeLimitedSimpleConsumerPool;
+import io.confluent.rest.exceptions.RestException;
+import io.confluent.rest.exceptions.RestServerErrorException;
+import kafka.api.PartitionFetchInfo;
+import kafka.cluster.Broker;
+import kafka.common.TopicAndPartition;
+import kafka.consumer.ConsumerConfig;
+import kafka.javaapi.FetchRequest;
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.message.ByteBufferMessageSet;
+import kafka.message.MessageAndMetadata;
+import kafka.message.MessageAndOffset;
+import kafka.serializer.Decoder;
+import kafka.serializer.DefaultDecoder;
+import kafka.utils.VerifiableProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SimpleConsumerManager {
+
+  private static final Logger log = LoggerFactory.getLogger(SimpleConsumerManager.class);
+
+  private final int maxPoolSize;
+  private final MetadataObserver mdObserver;
+  private final SimpleConsumerFactory simpleConsumerFactory;
+
+  private final Map<Broker, SimpleConsumerPool> simpleConsumersPools;
+
+  private AtomicInteger correlationId = new AtomicInteger(0);
+
+  private final Decoder<Object> avroDecoder;
+  private final Decoder<byte[]> binaryDecoder;
+
+  public SimpleConsumerManager(final KafkaRestConfig config,
+                               final MetadataObserver mdObserver,
+                               final SimpleConsumerFactory simpleConsumerFactory) {
+
+    this.mdObserver = mdObserver;
+    this.simpleConsumerFactory = simpleConsumerFactory;
+
+    maxPoolSize = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_POOL_SIZE_CONFIG);
+    simpleConsumersPools = new HashMap<Broker, SimpleConsumerPool>();
```

Comment:
I think this needs to be a ConcurrentHashMap or other code needs some synchronization. There could be many calls to `consume()` in parallel.